File: android/domain/src/main/kotlin/com/cacompte/domain/model/Participant.kt
package com.cacompte.domain.model

import kotlinx.serialization.Serializable
import java.util.UUID

/** Miroir de `Participant.swift`. `Participant.ID` côté Swift = `UUID` (via `Identifiable`) —
 * ici simplement `java.util.UUID`, pas de type enveloppe supplémentaire. */
@Serializable
data class Participant(
    @Serializable(with = UUIDSerializer::class)
    val id: UUID = UUID.randomUUID(),
    val displayName: String,
    val seatIndex: Int,
    /** Nul pour les jeux individuels ; une même valeur = même équipe (la valeur elle-même n'a
     * pas de sens au-delà de l'égalité). */
    val teamID: String? = null,
)

File: apple/QuiMeneKit/Sources/Domain/Model/Participant.swift
import Foundation

public struct Participant: Identifiable, Sendable, Codable, Hashable {
  public let id: UUID
  public let displayName: String
  public let seatIndex: Int
  /// Doc 05 « Belote » — `nil` pour tous les jeux individuels. Deux participants partagent
  /// une équipe s'ils portent le même `teamID` ; la valeur elle-même (ex. "A"/"B") n'a pas de
  /// sens en dehors de cette égalité.
  public let teamID: String?

  public init(id: UUID = UUID(), displayName: String, seatIndex: Int, teamID: String? = nil) {
    self.id = id
    self.displayName = displayName
    self.seatIndex = seatIndex
    self.teamID = teamID
  }
}
